src/Serialization/Compact.php

```diff
@@ -77,13 +89,19 @@ protected function parsePayload($payloadJson)
 
     /**
      * @param string $jwt
+     *
      * @return Token
+     * @throws \InvalidArgumentException
      */
     public function deserialize($jwt)
     {
         $token = new Token();
 
-        list($encodedHeader, $encodedPayload, $encodedSignature) = explode('.', $jwt);
+        if (empty($jwt)) {
+            throw new \InvalidArgumentException('Not a valid JWT string passed for deserialization');
+        }
+
+        list($encodedHeader, $encodedPayload, $encodedSignature) = array_pad(explode('.', $jwt, 3), 3, null);
```

Comment:
The spec describes a valid token as having "at least one period" at section [7.2.1](https://tools.ietf.org/html/draft-ietf-oauth-json-web-token-30#section-7.2).

A valid token is not required to provide the signature portion of the token. Manually adding a third (empty) value to the decoded token would not be a true representation of the token submitted for deserialization.

My preference would be to not pad the array to 3 entries. What do you think?